src/structlog/processors.py

```diff
@@ -360,13 +367,40 @@ def format_exc_info(
     If there is no ``exc_info`` key, the *event_dict* is not touched.
     This behavior is analogue to the one of the stdlib's logging.
     """
-    exc_info = event_dict.pop("exc_info", None)
-    if exc_info:
-        event_dict["exception"] = _format_exception(
-            _figure_out_exc_info(exc_info)
-        )
 
-    return event_dict
+    def __init__(
+        self,
+        exception_formatter: ProcessorExceptionFormatter = _format_exception,
+    ) -> None:
+        self.format_exception = exception_formatter
+
+    def __call__(
+        self, logger: WrappedLogger, name: str, event_dict: EventDict
+    ) -> EventDict:
+        exc_info = event_dict.pop("exc_info", None)
+        if exc_info:
+            event_dict["exception"] = self.format_exception(
+                _figure_out_exc_info(exc_info)
+            )
+
+        return event_dict
+
+
+format_exc_info = ExceptionFormatter()
+"""
+Replace an ``exc_info`` field with an ``exception`` string field:
+
+If *event_dict* contains the key ``exc_info``, there are two possible
```

Comment:
off by 1? ;)

Reply:
This is just a copy of *your* comment :-P
But I will fix it for you ;-)